src/ffi/mod.rs

```diff
@@ -131,6 +136,9 @@ pub extern "C" fn telio_new(
         TELIO_RES_OK => telio_log_debug!("Operation was succesful (telio_new)"),
     };
 
+    unsafe {
+        telio_log_info!("telio_new entry with instance id: {}. features: {:?}. Log level: {:?}. Event_ptr: {:?}. Logger_ptr: {:?}. Return value: {}", (*(*dev)).id, features, log_level, events, logger, ret);
```

Comment:
I think that if `telio_new_common` returns an error, this will dereference null pointer.

Reply:
This is not solved, you are now checking 'outer' pointer which is passed to us from applications (a place where we should store the pointer to telio instance) but you are not checking the inner pointer which we do set. Considering how error prone this is, please add a test that verifies this situation - that we will not crash if `telio_new_common` returns an error (and doesn't set the address where telio instance is located).